components/automate-ui/src/app/pages/project/rules/project-rules.component.html

```diff
@@ -1,6 +1,6 @@
 <chef-page
   subheading="Resources that match the type and all conditions are included in the project."
-  confirm-btn-text="Save Rule"
+  [attr.confirm-btn-text]="getConfirmBtnText()"
```

Comment:
bind the attribute to the getConfirmBtnText function

Reply:
A new pattern we started using just recently: when the function for generating a label is a simple expression (e.g. just 2 text phrases like this) then inline it to make it easier for the code reader. Thus:
```suggestion
  [attr.confirm-btn-text]="saving ? 'Saving Rule...' : 'Save Rule'"
```
Here's another example of its use: https://github.com/chef/automate/blob/f92742d7b5033b306ac2fb86cb6e75ea9bd0f60e/components/automate-ui/src/app/pages/project/confirm-apply-stop-modal/confirm-apply-stop-modal.component.html#L15

Of course, that means a little extra work for your tests: you will have to query the DOM.
But it is usually better to make opt for simpler code-under-test rather than simpler test.

Reply:
Thank you for the specific examples to reference